src/openassetio-core/include/openassetio/internal.hpp

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 The Foundry Visionmongers Ltd
+#pragma once
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+/**
+ * Contains private internal implementation details.
+ */
+namespace internal {
+namespace access {
+/**
+ * Common constant values for strong enumerations of access pattern.
+ *
+ * We must ensure correspondence between the values of the various
+ * workflows' strong enums. This allows for comparison, string lookup,
+ * simplifies serialisation, and supports language (especially C)
+ * bindings.
+ */
+enum Access : std::size_t { kRead = 0, kWrite, kCreateRelated };
+}  // namespace access
+
+namespace capability {
+/**
+ * Common constant values for strong enumerations of capability sets.
+ *
+ * We must ensure correspondence between the values of the middleware
+ * and manager implementation's strong enums. This allows for
+ * comparison, string lookup, simplifies serialisation, and supports
+ * language (especially C) bindings.
+ */
+enum Capability : std::size_t {
```

Comment:
[fix] Needs to be 'manager' qualified somehow - as we'll pretty certainly have 'host' capabilities at some point.

Reply:
OK good point, put in an addtional `manager` sub-namespace in 25bfd4c1.  

Can't simply change the name of the `enum`, since the values are exported to the namespace, so could potentially conflict with a future host enum.  Could switch to `enum class`, but that would mean adding lots more `static_cast`s everywhere.

Reply:
cool of the beans 👍 nice one.